packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -980,11 +976,33 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return _getTextPositionVertical(offset, verticalOffset);
   }
 
-  /// Keeping [selection]'s [TextSelection.baseOffset] fixed, move the
-  /// [TextSelection.extentOffset] down by one line.
+  // Deletes the current uncollapsed [selection].
+  void _deleteSelection(SelectionChangedCause cause) {
+    assert(_selection != null);
+    assert(_selection?.isCollapsed == false);
+
+    if (_readOnly || !_selection!.isValid || _selection!.isCollapsed) {
+      return;
+    }
+
+    final String text = textSelectionDelegate.textEditingValue.text;
+    final String textBefore = _selection!.textBefore(text);
```

Comment:
This could go wrong when the `TextEditingValue` has changed but `_selection` hasn't (e.g. when `textSelectionDelegate.textEditingValue.text` has already been changed by another key event took place earlier in the current frame). Maybe this method should operate based on `textSelectionDelegate.textEditingValue.selection` instead of `_selection` since you don't need to know the text layout to delete the current selection, and you want to delete the "current" selection instead of the selection from the last frame.

Reply:
I think this is one of the comments you told me to ignore, but just to be sure, should I leave this as it is for now?

Reply:
Yeah we can address this in a different PR.